dotnet/src/SemanticKernel.Abstractions/Events/SKEventArgs.cs

```diff
@@ -14,22 +14,27 @@ public abstract class SKEventArgs : EventArgs
     /// <summary>
     /// Initializes a new instance of the <see cref="SKEventArgs"/> class.
     /// </summary>
-    /// <param name="metadata">Function metadata</param>
+    /// <param name="function">Kernel function</param>
     /// <param name="context">Context related to the event</param>
-    internal SKEventArgs(SKFunctionMetadata metadata, SKContext context)
+    internal SKEventArgs(KernelFunction function, SKContext context)
     {
+        Verify.NotNull(function);
         Verify.NotNull(context);
-        Verify.NotNull(metadata);
 
-        this.FunctionMetadata = metadata;
+        this.Function = function;
         this.SKContext = context;
         this.Metadata = new();
     }
 
+    /// <summary>
+    /// Kernel function
+    /// </summary>
+    public KernelFunction Function { get; }
+
     /// <summary>
     /// Function metadata
     /// </summary>
-    public SKFunctionMetadata FunctionMetadata { get; }
+    public SKFunctionMetadata FunctionMetadata => this.Function.GetMetadata();
```

Comment:
Is this still needed? A caller that needs it can just do e.Function.GetMetadata().